src/czechpmdevs/buildertools/schematics/SchematicsManager.php

```diff
@@ -92,7 +92,7 @@ public static function loadSchematic(string $schematic, Closure $callback): void
 				return;
 			}
 
-			self::$loadedSchematics[$task->name] = $task->blockStorage->asBlockArray();
+			self::$loadedSchematics[$task->name] = igbinary_unserialize($task->blockStorage)->asBlockArray();
```

Comment:
Why the api class [NonThreadSafeValue](https://github.com/pmmp/PocketMine-MP/blob/major-next/src/thread/NonThreadSafeValue.php#L37) is not used?

Reply:
Same as igbinary_serialize:
> igbinary_* were recommended by Dylan (here https://discord.com/channels/373199722573201408/373214753147060235/1101162479129739384) and when Javier Leon talked about NonThreadSafeValue, dylan said "NonThreadSafeValue is convenient but it has a performance cost" (https://discord.com/channels/373199722573201408/373214753147060235/1101202901008908358)

Reply:
I would prefer to use NonThreadSafeValue. Schematic loading is not process called in loop, so the performance loss is negligible here.